src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { useState, FormEvent, FC } from 'react';
+import { Todo } from './types/Todo';
+
+enum DefaultField {
+  title = 'Enter a title',
+  select = 'Choose a user',
+}
+
+const preparedData = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(preparedData);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    setErrorTitle(!title.trim());
+    setErrorUser(!userId);
+
+    const formattedTitle = title.trim();
+
+    if (formattedTitle.length === 0 || !userId) {
+      return;
+    }
+
+    const ids = todos.map(({ id }) => id);
+    const id = Math.max(...ids, 0) + 1;
+
+    // const user = usersFromServer.find(someUser => someUser.id === userId);
+
+    // if (!user) {
+    //   setErrorUser(true);
+
+    //   return;
+    // }
+
+    const newTodo: Todo = {
+      id,
+      title,
+      userId,
+      completed: false,
+      user: usersFromServer.find(user => user.id === userId),
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+    reset();
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setErrorUser(false);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" onSubmit={handleSubmit}>
```

Comment:
The `action` attribute in the `form` element is unnecessary since form submission is handled by JavaScript. Removing this attribute will help avoid confusion.